NEXT_CHANGELOG.md

```diff
@@ -26,6 +26,19 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
 # [x.x.x] (unreleased) - 2022-mm-dd
 ## ❗ BREAKING ❗
 
+### improve plugin registration predictability ([PR #2181](https://github.com/apollographql/router/pull/2181))
+
+This replaces [ctor](https://crates.io/crates/ctor) with [linkme](https://crates.io/crates/linkme). `ctor` enables rust code to execute before `main`. This can be a source of undefined behaviour and we don't need our code to execute before `main`. `linkme` provides a registration mechanism that is perfect for this use case, so switching to use it makes the router more predictable, simpler to reason about and with a sound basis for future plugin enhancements.
+
+This is marked as a "breaking" change, because any existing rust plugins will need to add:
+
+```
+linkme = "0.3"
+```
+to their list of `Cargo.toml` dependencies. No router interfaces are incompatibly changed by this.
```

Comment:
Needs update, and to be moved to another section

Reply:
> (Note that ctor is still in the dependency graph, through `proteus` / `typetag` / `inventory`)

 It's still better to not use `ctor` if you don't need it, so steps in the right direction...

Reply:
done: https://github.com/apollographql/router/pull/2181/commits/57c557e1705707cd57c99097efa0dc13686817a0